### PR TITLE
[full-ci] Extend multiple space selection details

### PR DESCRIPTION
**Pull request by lookacat**
## Description
See https://github.com/owncloud/web/issues/8222

## Related Issue

- Works towards https://github.com/owncloud/web/issues/8222
- Fixes https://github.com/owncloud/web/issues/8246

## Screenshots (if appropriate):
Screenshot (29)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: packages/web-pkg/src/components/sideBar/Spaces/Details/SpaceDetailsMultiple.vue [hunk 1]
```diff
@@ -1,21 +1,121 @@
 <template>
-  <div class="oc-mt-xl">
-    <div class="oc-flex space-info">
-      <oc-icon name="layout-grid" size="xxlarge" />
-      <p v-translate>Multiple spaces selected</p>
+  <div id="oc-spaces-details-multiple-sidebar">
+    <div class="spaces-preview oc-mb">
+      <div class="spaces-preview-body">
+        <oc-icon class="preview-icon" size="xxlarge" variation="passive" name="layout-grid" />
+        <p class="preview-text" data-testid="selectedFilesText" v-text="selectedSpacesString" />
```
Comment (JammingBen) [inline packages/web-pkg/src/components/sideBar/Spaces/Details/SpaceDetailsMultiple.vue:6]: Let's either remove `data-testid` (as we don't use it in the tests), or rename it to `selectedSpacesText`. The same goes for the testids down below.

File: packages/web-pkg/src/components/sideBar/Spaces/Details/SpaceDetailsMultiple.vue [hunk 2]
```diff
@@ -1,21 +1,121 @@
...
+    }
+  },
+  setup(props) {
+    const { $ngettext, $gettextInterpolate } = useTranslations()
+    const totalSelectedSpaceQuotaTotal = computed(() => {
+      let total = 10
+      props.selectedSpaces.forEach((space) => {
+        total += space.spaceQuota.total
+      })
+      return filesize(total)
```
Comment (JammingBen) [inline packages/web-pkg/src/components/sideBar/Spaces/Details/SpaceDetailsMultiple.vue:48]: TIL: We have a nice `formatFileSize` helper 🙂 Let's use that. It needs `$language.current` as second param, you can get that from `useTranslations()`.

File: packages/web-pkg/tests/unit/components/sidebar/Spaces/Details/SpaceDetailsMultiple.spec.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  id: '1',
+  shareType: ShareTypes.space.value,
+  collaborator: {
+    onPremisesSamAccountName: 'Alice',
+    displayName: 'alice'
+  },
+  role: {
+    name: spaceRoleManager.name
+  }
+}
```
Comment (JammingBen) [inline packages/web-pkg/tests/unit/components/sidebar/Spaces/Details/SpaceDetailsMultiple.spec.ts:32]: This file looks like a lot of copy-paste, which is actually not needed here. You can probably remove `spaceShare` and all the store-related stuff in `createWrapper` (except for `defaultStoreMockOptions` maybe).

```suggestion
```

File: packages/web-pkg/tests/unit/components/sidebar/Spaces/Details/SpaceDetailsMultiple.spec.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  return {
+    wrapper: shallowMount(SpaceDetailsMultiple, {
+      global: {
+        plugins: [...defaultPlugins(), store],
+        directives: {
+          OcTooltip: jest.fn()
+        },
+        provide: {
+          displayedItem: spaceResource
+        }
```
Comment (JammingBen) [inline packages/web-pkg/tests/unit/components/sidebar/Spaces/Details/SpaceDetailsMultiple.spec.ts:61]:
```suggestion
```

File: packages/web-pkg/src/components/sideBar/Spaces/Details/SpaceDetailsMultiple.vue [hunk 5]
```diff
@@ -1,21 +1,121 @@
...
-      <p v-translate>Multiple spaces selected</p>
+  <div id="oc-spaces-details-multiple-sidebar">
+    <div class="spaces-preview oc-mb">
+      <div class="spaces-preview-body">
+        <oc-icon class="preview-icon" size="xxlarge" variation="passive" name="layout-grid" />
+        <p class="preview-text" data-testid="selectedFilesText" v-text="selectedSpacesString" />
+      </div>
+    </div>
+    <div>
+      <table class="details-table" :aria-label="detailsTableLabel">
```
Comment (JammingBen) [inline packages/web-pkg/src/components/sideBar/Spaces/Details/SpaceDetailsMultiple.vue:10]: `detailsTableLabel` does not exist yet.

**Review by JammingBen — CHANGES_REQUESTED** [review]

File: packages/web-pkg/src/components/sideBar/Spaces/Details/SpaceDetailsMultiple.vue [hunk 6]
```diff
@@ -1,21 +1,121 @@
...
+        <tr data-testid="filesCount">
+          <th scope="col" class="oc-pr-s" v-text="$gettext('Total:')" />
```
Comment (kulmann) [inline packages/web-pkg/src/components/sideBar/Spaces/Details/SpaceDetailsMultiple.vue:12]:
```suggestion
          <th scope="col" class="oc-pr-s" v-text="$gettext('Total quota:')" />
```

File: packages/web-pkg/src/components/sideBar/Spaces/Details/SpaceDetailsMultiple.vue [hunk 7]
```diff
@@ -1,21 +1,121 @@
...
+          <td v-text="totalSelectedSpaceQuotaTotal" />
+        </tr>
+        <tr data-testid="foldersCount">
+          <th scope="col" class="oc-pr-s" v-text="$gettext('Remaining:')" />
```
Comment (kulmann) [inline packages/web-pkg/src/components/sideBar/Spaces/Details/SpaceDetailsMultiple.vue:16]:
```suggestion
          <th scope="col" class="oc-pr-s" v-text="$gettext('Remaining quota:')" />
```

File: packages/web-pkg/src/components/sideBar/Spaces/Details/SpaceDetailsMultiple.vue [hunk 8]
```diff
@@ -1,21 +1,121 @@
...
+          <td v-text="totalSelectedSpaceQuotaRemaining" />
+        </tr>
+        <tr data-testid="spacesCount">
+          <th scope="col" class="oc-pr-s" v-text="$gettext('Used:')" />
```
Comment (kulmann) [inline packages/web-pkg/src/components/sideBar/Spaces/Details/SpaceDetailsMultiple.vue:20]:
```suggestion
          <th scope="col" class="oc-pr-s" v-text="$gettext('Used quota:')" />
```

**Review by kulmann — CHANGES_REQUESTED** [review]

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/31706/12/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31706/tracing/participant-management-alice-2023-1-17-09-33-17.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31706/tracing/participant-management-anonymous-2023-1-17-09-33-51.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31706/tracing/participant-management-brian-2023-1-17-09-33-26.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31706/tracing/participant-management-carol-2023-1-17-09-33-39.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31706/tracing/participant-management-marie-2023-1-17-09-33-30.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31706/tracing/participant-management-richard-2023-1-17-09-33-34.zip 


</p></details>

File: packages/web-pkg/src/components/sideBar/Spaces/Details/SpaceDetailsMultiple.vue [hunk 9]
```diff
@@ -1,21 +1,159 @@
...
+      return formatFileSize(used, $language.current)
+    })
+    const totalEnabledSpaces = computed(() => {
+      let enabled = 0
+      props.selectedSpaces.forEach((space) => {
+        if (!space.disabled) {
+          enabled++
+        }
+      })
+      return enabled
```
Comment (JammingBen) [inline packages/web-pkg/src/components/sideBar/Spaces/Details/SpaceDetailsMultiple.vue:85]: I'd prefer this for more readability:

```suggestion
      return props.selectedSpaces.filter((s) => !s.disabled).length
```

File: packages/web-pkg/src/components/sideBar/Spaces/Details/SpaceDetailsMultiple.vue [hunk 10]
```diff
@@ -1,21 +1,159 @@
...
+    })
+    const totalDisabledSpaces = computed(() => {
+      let disabled = 0
+      props.selectedSpaces.forEach((space) => {
+        if (space.disabled) {
+          disabled++
+        }
+      })
+      return disabled
```
Comment (JammingBen) [inline packages/web-pkg/src/components/sideBar/Spaces/Details/SpaceDetailsMultiple.vue:94]: ^ Same here.

**Review by JammingBen — APPROVED** [review]
LGTM 🥳 Minor nitpick, approving anyway since it's a "code-readability"-thing.

**Pull request merged**